linchpin/__init__.py

```diff
@@ -547,7 +549,8 @@ def get_pf_data_from_rundb(self, targets, run_id=None, tx_id=None):
         return pinfile
 
 
-    def do_action(self, provision_data, action='up', run_id=None, tx_id=None):
+    def do_action(self, provision_data, action='up', run_id=None, tx_id=None,
+                  flags={}):
```

Comment:
Why is flags necessary? See my other comments for details.

Reply:
@herlo My thoughts while implementation were there might be multiple flags coming up in future which might need to be accomodated . However, passing these as context variables seems much logical solution at this point.